book/src/css.md

```diff
@@ -0,0 +1,266 @@
+```
+
+This way, only the first button gets colored red.
+If you want to make sure that CSS rules only apply to a hand-selected set of nodes, adding your own style classes is your best bet.
+Just make sure that you do not choose a name for your style class that GTK either adds itself or provides CSS rules for.
+This brings us to the next point.
+
+<div style="text-align:center"><img src="img/css_3.png"/></div>
+
+### CSS Rules Provided by GTK
+
+Certain styles are common enough that GTK provides its own CSS rules for.
+If you want to indicate that your button leads to a destructive or suggested action, you can add the "destructive-action" or "suggested-action" style class to it.
+You do not have to provide a CSS file here.
+After you have added the style class, the `button` node will match the CSS rule provided by GTK.
+Most widgets will document these rules in their documentation under [CSS nodes](../docs/gtk4/struct.Button.html#css-nodes).
+
+<span class="filename">Filename: listings/css/4/main.rs</span>
+
+```rust ,no_run,noplayground
+{{#rustdoc_include ../listings/css/4/main.rs:buttons}}
+```
+
+<div style="text-align:center"><img src="img/css_4.png"/></div>
+
+### Interface Builder
+
+We can also add style classes with the interface builder.
+Just add the `<style>` element to your widget.
+The `<style>` element is documented [here](../docs/gtk4/struct.Widget.html#gtkwidget-as-gtkbuildable). 
+
+<span class="filename">Filename: listings/css/5/window/window.ui</span>
+
+```xml
+{{#rustdoc_include ../listings/css/5/window/window.ui}}
+```
+
+
+## Pseudo-classes
+
+Sometimes you want your CSS rules to apply to even more precise conditions than style classes allow.
+That is where [pseudo-classes](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes) come in.
+With pseudo-classes you can specify a specific state of the CSS node.
+Let us use a single button with the style class `button_1` added to demonstrate this concept.
+
+<span class="filename">Filename: listings/css/6/window/window.ui</span>
+
+```xml
+{{#rustdoc_include ../listings/css/6/window/window.ui}}
+```
+
+By adding the pseudo-class `hover`, we say that we only want this rule to apply for `button` nodes with style class `button_1` over which we currently hover the mouse pointer.
+We also set the `background-image` to `none` and add a bit of transition time.
+
+<span class="filename">Filename: listings/css/6/style.css</span>
+
+```css
+{{#rustdoc_include ../listings/css/6/style.css}}
+```
+
+If we now hover over the button, we see that over the span of one second its background turns yellow and its font turns red.
+After we removed the cursor, the button returns to its original state.
+
+<div style="text-align:center">
+ <video autoplay muted loop>
+  <source src="vid/css_6.webm" type="video/webm">
+Your browser does not support the video tag.
+ </video>
+</div>
+
+## Nodes
+
+In the previous examples a widget always corresponded to a single CSS node.
```

Comment:
Ok, nevermind another comment I guess. :)

Reply:
That said, not sure if mentioning GtkComboBox, well, anywhere in the book is a good idea - hopefully that widget is deprecated sooner rather than later.

Reply:
I will replace it with GtkMenuButton and then manipulate one if its child nodes